spec/javascripts/directives/scheduler/updateDropdownForFilter_spec.js

```diff
@@ -6,17 +6,17 @@ describe('update-drop-down-for-filter initialization', function() {
     var element = angular.element(
       '<form name="angularForm">' +
       '<select id="filter_typ" name="filter_typ" ng-model="scheduleModel.filter_typ"><option>All VMs</option><option>Ketchup</option><option>Relish</option></select>' +
-      '<select id="filter_value" name="filter_value" update-dropdown-for-filter dropdown-model="scheduleModel" dropdown-list="filterList" filter-hide="filterValuesEmpty" ng-model="scheduleModel.filter_value"><option value="" label="Choose">Choose</option><option value="0" label="Mustard">Mustard</option><option value="1" label="Ketchup">Ketchup</option><option value="2" label="Relish">Relish</option>' +
+      '<select id="filter_value" name="filter_value" update-dropdown-for-filter dropdown-model="scheduleModel" dropdown-list="filterList" filter-hide="filterValuesEmpty" ng-model="scheduleModel.filter_value"><option disabled="" value="" class="" selected="selected">Choose</option><option value="string:Mustard" label="Mustard">Mustard</option><option value="string:Ketchup" label="Ketchup">Ketchup</option><option value="string:Relish" label="Relish">Relish</option>' +
```

Comment:
@AparnaKarve I'm sorry, you've misunderstood the article. The code sample merely illustrates what `ng-options` does when used. That has indeed changed, but it's not something you would ever use by hand. (Also, `track by` is not a new thing in 1.4, I think it's actually been there from at least 1.2 but definitely 1.3.)

What it means is that when you *do* use ng-options, **previously** it would generate input values by stringifying the array item .. that is, with `ng-options="i in [ 1, 2, 3 ]"`, it would create `<option value="1">1</option>` (and so on). That was a problem, because you would get the same option value for `1` and `"1"`. What changed in angular 1.4 is that we use strict comparison, meaning 1 can no longer be equal to "1". Thus, what ng-options does **now** is that it serializes both the type and the value to the option value.

But all that is really internal to ng-options (it adds a parser/formatter pair on the model controller to convert back and forth), which you're not using in the spec.